File: python/hackymess.py
def meme(thing):
	chars = []
	for char in thing:
		if chars[-1:] and chars[-1]["char"] == char:
			chars[-1]["count"] += 1
		else:
			chars.append({ "char": char, "count": 1 })
	final = ""
	for data in chars:
		final += str(data["count"]) + data["char"]
	return final

print(meme(raw_input("Type some crap here: ")))

File: js/hackymess.js
"use strict";

function meme(thing) {
	var chars = [];
	for (var i = 0; i < thing.length; i++) {
		if (chars[chars.length - 1] && chars[chars.length - 1].char === thing.charAt(i)) {
			chars[chars.length - 1].count++;
		} else {
			chars.push({ char: thing.charAt(i), count: 1 });
		}
	}
	var final = "";
	for (var j = 0; j < chars.length; j++) {
		final += chars[j].count + chars[j].char;
	}
	return final;
}

alert(meme(prompt("Type some crap here")));